lib/enum/basic_enum.rb

```diff
@@ -85,66 +98,78 @@ def element_frequency_count
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.find_all {|ele| ele.end_with? 'er'}
   end
 
 
   # Refer to find_index
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-index
   def index_of_first_awesome_element
+    collection.find_index {|person| person[:awesome] == true}
   end
 
   # Refer to group_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-group-by
   def group_elements_by_favorite_language
+    collection.group_by {|person| person[:favorite_language]}
   end
 
   # Refer to inject / reduce
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-inject
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-reduce
   def sum_of_experiences
+    collection.map {|person| person[:years_experience]}.inject(:+)
   end
 
   # Refer to inject / reduce
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-inject
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-reduce
   def longest_element_name_using_inject
+    collection.max_by {|person| person[:name].length}
   end
 
   # Refer to max_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-max-by
   def most_experienced_element
+    collection.max_by {|person| person[:years_experience]}
   end
 
   # Refer to include? / member?
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-include-3F
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-member-3F
   def element_present?(name)
+    collection.member?(name)
   end
 
   # Refer to minmax_by
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-minmax_by
   def elements_with_longest_and_shortest_names
+    collection.minmax_by {|person| person[:name].length}
   end
 
   # Refer to partition
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-partition
   def separate_elements_that_like_functional_programming_from_rest
+    collection.partition {|person| person[:likes_functional_programming] == true}
   end
 
   # Refer to reject
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-reject
   def elements_who_dislike_functional_programming
+    collection.reject {|person| person[:likes_functional_programming] == true}
   end
 
   # Refer to sort
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-sort
   # You will need to use the 'spaceship' operator <=>
   def elements_sorted_by_experience
+    collection.sort {|exp1, exp2| exp1[:name][:years_experience] <=> exp2[:name][:years_experience]}
```

Comment:
Almost there. First, it's more accurate to call the local variables `|person1, person2|` rather than `|exp1, exp2|`. Second, if you have `person1[:name][:years_experience]` that would imply you're looking for experience nested inside the name which is itself nested inside person1. That's not actually true, since name and years of experience are on the same "level." So you can just do `person1[:years_experience]`. 

Putting it all together: 

`collection.sort {|person1, person2| person1[:years_experience] <=> person2[:years_experience]}`

Make sure to run your tests every time you write a method!